lib/src/component_declaration/component_base.dart

```diff
@@ -376,6 +377,261 @@ abstract class UiComponent<TProps extends UiProps> extends react.Component imple
+
+        throw new PropError.required(prop.key, prop.errorMessage);
+      });
+    });
+  }
+
+  /// Returns a new ClassNameBuilder with className and blacklist values added from [CssClassPropsMixin.className] and
+  /// [CssClassPropsMixin.classNameBlacklist], if they are specified.
+  ///
+  /// This method should be used as the basis for the classNames of components receiving forwarded props.
+  ///
+  /// > Should be used alongside [copyUnconsumedProps] or [copyUnconsumedDomProps].
+  ClassNameBuilder forwardingClassNameBuilder() {
+    return new ClassNameBuilder.fromProps(this.props);
+  }
+
+  @override
+  @mustCallSuper
+  void componentWillReceiveProps(Map nextProps) {
+    if (inReactDevMode) {
+      validateProps(nextProps);
+    }
+  }
+
+  @override
+  @mustCallSuper
+  void componentWillMount() {
+    if (inReactDevMode) {
+      validateProps(props);
+    }
+  }
+
+  @override
+  @mustCallSuper
+  void componentWillUnmount() {
+    _disposableProxy?.dispose();
+  }
+
+
+  // ----------------------------------------------------------------------
+  // ----------------------------------------------------------------------
+  //   BEGIN Typed props helpers
+  //
+
+  /// A typed props object corresponding to the current untyped props Map ([unwrappedProps]).
+  ///
+  /// Created using [typedPropsFactory] and cached for each Map instance.
+  // FIXME explain that this needs to be concrete in Dart 2 for soundness; link to issue
+  @override
+  TProps get props {
+    throw new UngeneratedError();
+  }
+
+  /// Equivalent to setting [unwrappedProps], but needed by react-dart to effect props changes.
+  @override
+  set props(Map value) => super.props = value;
+
+  /// The props Map that will be used to create the typed [props] object.
+  Map get unwrappedProps => super.props;
+  set unwrappedProps(Map value) => super.props = value;
+
+  /// Returns a typed props object backed by the specified [propsMap].
+  ///
+  /// Required to properly instantiate the generic [TProps] class.
+  TProps typedPropsFactory(Map propsMap);
+
+  /// Returns a typed props object backed by the specified [propsMap].
+  ///
+  /// Required to properly instantiate the generic [TProps] class.
+  ///
+  /// This should be used where possible over [typedPropsFactory] to allow for
+  /// more efficient dart2js output.
+  TProps typedPropsFactoryJs(JsBackedMap propsMap);
+
+  /// Returns a typed props object backed by a new Map.
+  ///
+  /// Convenient for use with [getDefaultProps].
+  TProps newProps() => typedPropsFactoryJs(new JsBackedMap());
+
+  //
+  //   END Typed props helpers
+  // ----------------------------------------------------------------------
+  // ----------------------------------------------------------------------
+
+  // ----------------------------------------------------------------------
+  // ----------------------------------------------------------------------
+  //   BEGIN DisposableManagerV7 interface implementation
```

Comment:
Should we update `UiComponent2` to implement a more up-to-date version of this interface?

Reply:
Apparently DisposableManagerV7 is the latest 🤷‍♂️